### PR TITLE
Add more tests for uv lock

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Misc. cases we're missing vis-a-vis pip install.

### CODE REVIEW

File: crates/uv/tests/lock.rs [hunk 1]
```diff
@@ -5221,7 +5228,190 @@ fn lock_redact() -> Result<()> {
...
+        version = "0.1.0"
+        source = { editable = "." }
+        dependencies = [
+            { name = "uv-private-pypackage" },
+        ]
+
+        [[package]]
+        name = "uv-private-pypackage"
+        version = "0.1.0"
+        source = { git = "https://***@github.com/astral-test/uv-private-pypackage#d780faf0ac91257d4d5a4f0c5a0e4509608c0071" }
```
Comment (charliermarsh) [inline crates/uv/tests/lock.rs:5307]: We may need to revisit this, I'm not sure.

Comment (zanieb) [reply crates/uv/tests/lock.rs:5306]: We're writing the token to the lockfile?

Comment (charliermarsh) [reply crates/uv/tests/lock.rs:5306]: Yeah. If we don't, how would the user be able to provide it in the future?

Comment (charliermarsh) [reply crates/uv/tests/lock.rs:5306]: It differs from registries in that way... With registries, there's a mechanism for providing credentials at install-time.

Comment (zanieb) [reply crates/uv/tests/lock.rs:5330]: I think your Git credentials can be fetched from the system keychain (or whatever store Git is configured to use). It's tough when someone hands us credentials directly... but it seems bad to make it easy to commit them to VCS in a generated file.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs:5330]: Yeah I'm not sure what's right here. It feels like a footgun either way (e.g., if we omit, then `lock` followed by `sync` fails).

**Pull request merged**